gcloud-java-dns/src/main/java/com/google/gcloud/dns/Zone.java

```diff
@@ -33,20 +34,87 @@
  * @see <a href="https://cloud.google.com/dns/zones/">Google Cloud DNS managed zone
  * documentation</a>
  */
-public class Zone implements Serializable {
+public class Zone extends ZoneInfo {
 
-  // TODO(mderka) Zone and zoneInfo to be merged. Opened issue #605.
+  private static final long serialVersionUID = 564454483894599281L;
+  private transient Dns dns;
```

Comment:
Also, I think we should already have the Serialization test which should include Zone (as well as ZoneInfo).